src/mc_tasks/lipm_stabilizer/StabilizerTask.cpp

```diff
@@ -1135,219 +1133,230 @@ void StabilizerTask::distributeCoPonHorizon(const std::vector<Eigen::Vector2d> &
     modeledFzLeft_ = measuredFzLeft;
     modeledFzRight_ = measuredFzRight;
   }
-  double t_delay = clamp((c_.delayCoP - (t_ - tComputation_)), 0, c_.delayCoP);
-
-  Eigen::Vector3d targetForceLeft = Eigen::Vector3d::Zero();
-  Eigen::Vector3d targetForceRight = Eigen::Vector3d::Zero();
-
-  Eigen::Vector2d measuredLeftCoP_delayed;
-  double measuredFzLeft_delayed = 0;
-
-  measuredLeftCoP_delayed.x() = measuredLeftCoP.x() * exp(-c_.copFzLambda.x() * t_delay)
-                                + (1 - exp(-c_.copFzLambda.x() * t_delay)) * delayedTargetCoPLeft_.x();
-  measuredLeftCoP_delayed.y() = measuredLeftCoP.y() * exp(-c_.copFzLambda.y() * t_delay)
-                                + (1 - exp(-c_.copFzLambda.y() * t_delay)) * delayedTargetCoPLeft_.y();
+  const double t_delay = clamp((c_.delayCoP - (t_ - tComputation_)), 0, c_.delayCoP);
 
-  Eigen::Vector2d measuredRightCoP_delayed;
-  double measuredFzRight_delayed = 0;
-  measuredRightCoP_delayed.x() = measuredRightCoP.x() * exp(-c_.copFzLambda.x() * t_delay)
-                                 + (1 - exp(-c_.copFzLambda.x() * t_delay)) * delayedTargetCoPRight_.x();
-  measuredRightCoP_delayed.y() = measuredRightCoP.y() * exp(-c_.copFzLambda.y() * t_delay)
-                                 + (1 - exp(-c_.copFzLambda.y() * t_delay)) * delayedTargetCoPRight_.y();
-
-  if(supportFoot_ == ContactState::Left)
+  if(newCoPHorizonRef_)
```

Comment:
I think it would improve readability if we can avoid the introduction of this huge scope, can you either:
- introduce an helper function (prefered)
```
if(horizonRefUpdated_)
{
  doTheThing();
}
```
- invert the condition and introduce a goto
```
if(!horizonRefUpdated_)
{
  goto update_for_logging;
}
// do the thing
```